python/fapolicy_analyzer/ui/state_manager.py

```diff
@@ -83,17 +88,82 @@ def __update_dirty_queue(self):
         Returns True is there are unapplied changes, False otherwise."""
         if self.bDirtyQ != self.is_dirty_queue():
             self.bDirtyQ = self.is_dirty_queue()
-            self.changeset_queue_updated()
+            self.ev_changeset_queue_updated()
         return self.bDirtyQ
 
+    # ##################### Queue conversion utilities #####################
     def get_path_action_list(self):
         # Iterate through the StateManagers Changeset list
         # Each changeset contains a dict with at least one Path/Action pair
+        # The rationale for the tuple format is the Gtk TreeView widget display
         return [t for e in self.listChangeset for t in e.get_path_action_map().items()]
 
-    def add_system_notification(
-        self, notification: str, notification_type: NotificationType
-    ):
+    def __path_action_list_2_queue(self, listPathAction):
+        """Converts a list of Path/Action tuples to populate changeset queue
+This is a utility function intended for unit testing and troubleshooting
+the StateManager and its interactions with interfacing objects. It is used to
+populate the internal queue structure with Changeset objects, however it does
+not exercise the full normal end to end data path which will normally apply and
+deploy these changesets.
+"""
+        for e in listPathAction:
+            cs = Changeset()
+            if e[1] == "Add":
+                cs.add_trust(e[0])
+            elif e[1] == "Del":
+                cs.del_trust(e[0])
+            else:
+                print("Error: Path/Action Pair: "
+                      "Unknown Action: {} {}".format(e[0], e[1]))
+                continue
+            self.add_changeset_q(cs)
```

Comment:
Is this method being used anywhere?  The only calls I see to it are in the tests.

Reply:
It is a utility for packing the queue as you noted.

Path/Action tuples became the de facto standard structure for working with the TreeViews in the Gtk UI library. Dicts serialize easier. And the queue itself contains Changesets. I've got a ticket open to migrate away from Changesets in the queue. Why propagate the dependency, when the StateManager doesn't interact with the rust interface layer. It worked for prior tickets but it's too opaque now imo.

I did make a quick stab at a single controller (while you were on vaca), but It really was just adding another layer between the only client that would be using those interface functions (almost) so I tried to just change my frame of reference to something like, each tab is it's own application. Sorta, rest/endpoint of approach, because the ATDA maintains the current data in the TreeStore associated with the trust list object afaict.